includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1687,11 +1687,12 @@ public function process_payment_for_order( $cart, $payment_information, $schedul
 						// Include a new nonce for update_order_status to ensure the update order
 						// status call works when a guest user creates an account during checkout.
 						'redirect'       => sprintf(
-							'#wcpay-confirm-%s:%s:%s:%s',
+							'#wcpay-confirm-%s:%s:%s:%s:%s',
 							$payment_needed ? 'pi' : 'si',
 							$order_id,
 							$client_secret,
-							wp_create_nonce( 'wcpay_update_order_status_nonce' )
+							wp_create_nonce( 'wcpay_update_order_status_nonce' ),
+							$intent->get_payment_method_type(),
```

Comment:
This all looks good. I just started to think about sanitizing the values here in general. It's not only related to this PR. We have e.g. `$client_secret` that's being added in its raw version. 

It's interesting that the checkout uses [the default wp_redirect](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-checkout.php#L1094) and not the safe redirect which [would have the URL sanitization](https://github.com/WordPress/wordpress-develop/blob/6.7/src/wp-includes/pluggable.php#L1528-L1531) so I started to wonder if we need to sanitize the URL going forward.